### PR TITLE
Minor tests improvements for eth exec layer

### PR DESCRIPTION
**Pull request by AKorpusenko**
## Description
- Refactored `task_executor_test`. Added mock controller usage instead logs validation
- Added comments to `event_hanlder` tests

### CODE REVIEW

File: eth/eventhandler/event_handler_test.go [hunk 1]
```diff
@@ -345,6 +358,58 @@ func TestHandleBlockEventsStream(t *testing.T) {
...
+		eventsCh := make(chan executionclient.BlockLogs)
+
+		go func() {
+			defer close(eventsCh)
+			eventsCh <- block
+		}()
+
+		lastProcessedBlock, err := eh.HandleBlockEventsStream(eventsCh, false)
+		require.Equal(t, uint64(0x9), lastProcessedBlock)
+		require.NoError(t, err)
```
Comment (y0sher) [inline eth/eventhandler/event_handler_test.go:411]: I guess here other than checking the block number we could check that we are trying to execute the correct tasks?

**Review by y0sher — CHANGES_REQUESTED** [review]
Looks good generally, please take a look at my comments and fix formatting so we can merge.

File: eth/eventhandler/task_executor_test.go [hunk 2]
```diff
@@ -80,58 +81,58 @@ func TestExecuteTask(t *testing.T) {
...
 		task := NewReactivateClusterTask(eh.taskExecutor, ethcommon.HexToAddress("0x1"), []uint64{1, 2, 3}, shares)
+		validatorCtrl.EXPECT().ReactivateCluster(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
 		require.NoError(t, task.Execute())
-		require.NotZero(t, observedLogs.Len())
-		entry := observedLogs.All()[len(observedLogs.All())-1]
-		require.Equal(t, "reactivated cluster", entry.Message) // TODO: fix
-		require.Equal(t, int64(1), entry.ContextMap()["cluster_validators"])
+
+		//require.NotZero(t, observedLogs.Len())
```
[the remark this answers is not in the document]
Comment (AKorpusenko) [reply eth/eventhandler/task_executor_test.go]: Removed ✔️

File: eth/eventhandler/event_handler_test.go (hunk 1 above)
Comment (AKorpusenko) [reply eth/eventhandler/event_handler_test.go:411]: You're right. Working on it

**Comment by y0sher** [comment]
I rebased [tests/fix-eth-eventhandler-tests](https://github.com/bloxapp/ssv/tree/tests/fix-eth-eventhandler-tests) on top of stage so it'll be easier to see changes.

File: eth/eventhandler/event_handler_test.go [hunk 3]
```diff
@@ -99,21 +104,22 @@ func TestHandleBlockEventsStream(t *testing.T) {
...
-	operatorPrivKey, operatorPubKey := blskeygen.GenBLSKeyPair()
-
-	// Generate validator shares
-	validatorShares, _, err := newValidator()
-	require.NoError(t, err)
-	// Encode shares
-	encodedValidatorShares, err := validatorShares.Encode()
+	ops, err := createOperators(4)
 	require.NoError(t, err)
+	data, err := generateSharesData(ops)
```
Comment (y0sher) [inline eth/eventhandler/event_handler_test.go:109]: lets check for error here as well

File: eth/eventhandler/event_handler_test.go [hunk 4]
```diff
@@ -225,14 +226,58 @@ func TestHandleBlockEventsStream(t *testing.T) {
...
+			eventsCh <- block
+		}()
+
+		lastProcessedBlock, err = eh.HandleBlockEventsStream(eventsCh, false)
+		require.Equal(t, blockNum+1, lastProcessedBlock)
+		require.NoError(t, err)
+		blockNum++
+		// Check that validator was not registered, but nonce was bumped even event is malformed!
+		shares = eh.nodeStorage.Shares().List(nil)
+		require.Equal(t, 1, len(shares))
```
[the remark this answers is not in the document]
Comment (y0sher) [reply eth/eventhandler/event_handler_test.go:272]: we could also check it after previous event

File: eth/eventhandler/event_handler_test.go [hunk 5]
```diff
@@ -448,67 +547,148 @@ func simTestBackend(testAddr ethcommon.Address) *simulator.SimulatedBackend {
...
+	sec *bls.SecretKey
+	pub *bls.PublicKey
 }
 
-func generateRandomValidatorShare(splitKeys map[uint64]*bls.SecretKey) (*ssvtypes.SSVShare, *bls.SecretKey) {
+//func blsID(opID uint64) bls.ID { //yes, bls.ID is just a struct wrapping a byte array
+//	var id bls.ID
+//	id.SetDecString(fmt.Sprintf("%d", opID))
+//	return id
+//}
```
Comment (y0sher) [inline eth/eventhandler/event_handler_test.go:586]: remove this

File: eth/eventhandler/event_handler_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (y0sher) [reply eth/eventhandler/event_handler_test.go:272]: Also lets make sure this case is covered first - https://github.com/bloxapp/ssv/pull/1087 after that we can merge something basic and start testing for all other cases.

**Review by y0sher — CHANGES_REQUESTED** [review]
Some comments are probably leftovers from my code, but lets fix this and keep going.

File: eth/eventhandler/event_handler_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (AKorpusenko) [reply eth/eventhandler/event_handler_test.go]: Added nonce checking now :)

[the remark this answers is not in the document]
Comment (AKorpusenko) [reply eth/eventhandler/event_handler_test.go]: Regarding to the more complex test of #1087, I suggest such test scenario we've discussed with @moshe-blox:
- make the EventHandler think that it's operator ID is 4 (by passing it OperatorData interface that returns ID 4 and some whatever public key)
- register a validator with invalid share for operator 3, check that its saved to DB (operator 4 shouldnt check this share so it's considered valid)
- register a validator with invalid share for opeator 4, check that its not saved to DB
- register a validator with all shares valid, check thats its saved to DB

**Pull request merged**